src/sentry/discover/models.py

```diff
@@ -40,6 +40,7 @@ class DiscoverSavedQuery(Model):
     date_updated = models.DateTimeField(auto_now=True)
     visits = BoundedBigIntegerField(null=True, default=1)
     last_visited = models.DateTimeField(null=True, default=timezone.now)
+    is_default = models.BooleanField(null=True)
```

Comment:
Should you add a `unique_together` on is_default & created_by

Reply:
nvm, won't work since is_default=False would count too, maybe validate on the [save signal](https://docs.djangoproject.com/en/4.1/ref/signals/#django.db.models.signals.pre_save)?

Reply:
You can define conditional indexes in postgres. For example you could have an index on `organization_id, is_default=True`. You can define this kind of index in django with

```python
        constraints = [
            models.UniqueConstraint(
                fields=["organization_id"],
                name="unique_organization_default_query",
                condition=models.Q(is_default=True),
            ),
        ]
```

With this kind of index though, you'll need to move the is_default flag around within a single transaction or postgres will get salty.

Reply:
Thanks for the suggestion @markstory! I had just discovered constraints as you responded.

I'm a little unclear on what your last statement implies for things to watch out for or if there's anything more I need to do in this PR to handle the flag. Do you mean that if there are updates to `is_default`, the transaction needs to be marked atomic?

Reply:
Ah, I understand now. Thanks Mark!